### ISSUE TITLE
Error in Construct GEMs

### ISSUE THREAD

**Issue by yixudong** [axelalmet/flowsig#6]
Dear authors, I have encountered an error when building onstruct GEMs. I would appreciate your reply.

Firstly, this my scanpy workflow:

![image](https://github.com/user-attachments/assets/759be629-bfe8-4128-9db3-367aaa792be7)
![image](https://github.com/user-attachments/assets/fd70bf93-7272-4030-9417-e25a92d6055a)
![image](https://github.com/user-attachments/assets/b3a983e4-eaa2-4bd2-8439-65cad4c0edb0)
it seems there is nothing wrong with that, but when i run FlowSig im my windows 11 terminal: 
 
![image](https://github.com/user-attachments/assets/d95ca4dc-9585-4f7c-88f8-9bd22a0e8b71)
 it report an error: Raw data must have both row (cell) and column (gene) names.
But this part of the information seems to be in adata.obs and adata.obs, I don't know how to use it:
![image](https://github.com/user-attachments/assets/55f67467-2f34-4338-8b97-a2f0b7286ace)

I would be grateful if the author could solve my confusion！

(flowsigenv2) C:\Users\1\flowsig>pip list
Package                      Version
---------------------------- -----------
absl-py                      2.1.0
adjustText                   1.2.0
aiobotocore                  2.5.4
aiohappyeyeballs             2.4.0
aiohttp                      3.10.5
aioitertools                 0.12.0
aiosignal                    1.3.1
anndata                      0.10.9
annotated-types              0.7.0
annoy                        1.17.3
array_api_compat             1.8
asciitree                    0.3.3
asttokens                    2.4.1
astunparse                   1.6.3
attrs                        24.2.0
biothings-client             0.3.1
bleach                       6.1.0
bokeh                        3.4.3
botocore                     1.31.17
cachetools                   5.5.0
causaldag                    0.1a163
certifi                      2024.8.30
charset-normalizer           3.3.2
click                        8.1.7
cloudpickle                  3.0.0
cmake                        3.30.2
colorama                     0.4.6
colorcet                     3.1.0
conditional-independence     0.1a6
contourpy                    1.3.0
cycler                       0.12.1
dask                         2024.8.2
dask-expr                    1.1.13
dask-image                   2024.5.3
dataclasses                  0.6
datashader                   0.16.3
decorator                    5.1.1
dill                         0.3.8
distributed                  2024.8.2
dm-tree                      0.1.8
docopt                       0.6.2
docrep                       0.3.2
einops                       0.8.0
et-xmlfile                   1.1.0
executing                    2.1.0
fasteners                    0.19
filelock                     3.15.4
flatbuffers                  24.3.25
flowsig                      0.1.0
fonttools                    4.53.1
frozendict                   2.4.4
frozenlist                   1.4.1
fsspec                       2023.6.0
ftpretty                     0.4.0
gast                         0.6.0
geopandas                    1.0.1
goatools                     1.4.12
google-ai-generativelanguage 0.6.6
google-api-core              2.19.2
google-api-python-client     2.144.0
google-auth                  2.34.0
google-auth-httplib2         0.2.0
google-generativeai          0.7.2
google-pasta                 0.2.0
googleapis-common-protos     1.65.0
graphical-model-learning     0.1a8
graphical-models             0.1a21
grpcio                       1.66.1
grpcio-status                1.62.3
h5py                         3.11.0
h5sparse                     0.1.0
holoviews                    1.19.1
httplib2                     0.22.0
idna                         3.8
igraph                       0.11.6
imageio                      2.35.1
inflect                      7.3.1
ipdb                         0.13.13
ipython                      8.27.0
jedi                         0.19.1
Jinja2                       3.1.4
jmespath                     1.0.1
joblib                       1.4.2
keras                        3.5.0
kiwisolver                   1.4.7
lazy_loader                  0.4
legacy-api-wrap              1.4
leidenalg                    0.10.2
libclang                     18.1.1
linkify-it-py                2.0.3
llvmlite                     0.43.0
locket                       1.0.0
louvain                      0.8.2
Markdown                     3.7
markdown-it-py               3.0.0
MarkupSafe                   2.1.5
matplotlib                   3.9.2
matplotlib-inline            0.1.7
matplotlib-scalebar          0.8.1
mdit-py-plugins              0.4.1
mdurl                        0.1.2
mizani                       0.11.4
ml-dtypes                    0.4.0
more-itertools               10.5.0
mpmath                       1.3.0
msgpack                      1.0.8
multidict                    6.0.5
multipledispatch             1.0.0
multiscale_spatial_image     1.0.1
mygene                       3.2.2
namex                        0.0.8
natsort                      8.4.0
networkx                     3.3
numba                        0.60.0
numcodecs                    0.13.0
numexpr                      2.10.1
numpy                        1.26.4
ome-zarr                     0.9.0
omnipath                     1.0.8
openpyxl                     3.1.5
opt-einsum                   3.3.0
optree                       0.12.1
packaging                    24.1
pandas                       2.2.2
panel                        1.4.5
param                        2.1.1
parso                        0.8.4
partd                        1.4.2
patsy                        0.5.6
pgmpy                        0.1.26
pillow                       10.4.0
PIMS                         0.7
pip                          24.2
platformdirs                 4.2.2
plotnine                     0.13.6
pooch                        1.8.2
progressbar2                 4.5.0
prompt_toolkit               3.0.47
proto-plus                   1.24.0
protobuf                     4.25.4
psutil                       6.0.0
pure_eval                    0.2.3
pyarrow                      17.0.0
pyasn1                       0.6.0
pyasn1_modules               0.4.0
pyct                         0.5.0
pydantic                     2.9.0
pydantic_core                2.23.2
pydot                        3.0.1
pygam                        0.9.1
Pygments                     2.18.0
pyliger                      0.2.3
pynndescent                  0.5.13
pyogrio                      0.9.0
pyparsing                    3.1.4
pyproj                       3.6.1
python-dateutil              2.9.0.post0
python-igraph                0.11.6
python-utils                 3.8.2
pytz                         2024.1
pyviz_comms                  3.0.3
PyYAML                       6.0.2
requests                     2.32.3
rich                         13.8.0
rsa                          4.9
s3fs                         2023.6.0
scanpy                       1.10.2
scikit-image                 0.24.0
scikit-learn                 1.5.1
scipy                        1.11.4
seaborn                      0.13.2
session_info                 1.0.0
setuptools                   74.1.2
shapely                      2.0.6
six                          1.16.0
slicerator                   1.1.0
sortedcontainers             2.4.0
spatial-factorization        0.0.1
spatial_image                1.1.0
spatialdata                  0.2.2
squidpy                      1.6.1
stack-data                   0.6.3
statsmodels                  0.14.2
stdlib-list                  0.10.0
sympy                        1.13.2
tblib                        3.0.0
tensorboard                  2.17.1
tensorboard-data-server      0.7.2
tensorflow                   2.17.0
tensorflow-intel             2.17.0
tensorflow-probability       0.24.0
termcolor                    2.4.0
texttable                    1.7.0
tf_keras                     2.17.0
threadpoolctl                3.5.0
tifffile                     2024.8.30
toolz                        0.12.1
torch                        2.4.1
tornado                      6.4.1
tqdm                         4.66.5
traitlets                    5.14.3
typeguard                    4.3.0
typing                       3.7.4.3
typing_extensions            4.12.2
tzdata                       2024.1
uc-micro-py                  1.0.3
umap-learn                   0.5.6
uritemplate                  4.1.1
urllib3                      1.26.20
validators                   0.34.0
wcwidth                      0.2.13
webencodings                 0.5.1
Werkzeug                     3.0.4
wheel                        0.44.0
wrapt                        1.16.0
xarray                       2024.7.0
xarray-dataclasses           1.8.0
xarray-datatree              0.0.14
xarray-schema                0.0.3
xarray-spatial               0.4.0
xgboost                      2.1.1
XlsxWriter                   3.2.0
xyzservices                  2024.9.0
yarl                         1.9.11
zarr                         2.18.3
zict                         3.0.0

**Comment by axelalmet** [comment]
Hi yixudong,

Ah yes, I remember this error being an artefact from using pyLIGER. To fix it, if I recall correctly, try setting something like:

```
adata.obs.index.name = 'barcode'
adata.var.index.name = 'gene_name'
```

Let me know if that works!

Best wishes,
Axel.

**Comment by yixudong** [comment]
> Hi yixudong,
> 
> Ah yes, I remember this error being an artefact from using pyLIGER. To fix it, if I recall correctly, try setting something like:
> 
> ```
> adata.obs.index.name = 'barcode'
> adata.var.index.name = 'gene_name'
> ```
> 
> Let me know if that works!
> 
> Best wishes, Axel.


Dear Axel, 

it worked. Thank you very much.

Best wishes, Yi xudong.

**Comment by axelalmet** [comment]
Great! Closing this as the issue is fixed.
